shadowsocks/replay_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2020 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shadowsocks
+
+import (
+	"math/rand"
+	"testing"
+)
+
+func MakeVecs(n int) [][32]byte {
+	vecs := make([][32]byte, n)
+	for i := 0; i < n; i++ {
+		rand.Read(vecs[i][:])
+	}
+	return vecs
+}
+
+func TestIVCache_Active(t *testing.T) {
+	vecs := MakeVecs(2)
```

Comment:
Maybe add a comment saying that this assumes both vectors are different.

Reply:
Maybe a better approach is to just hardcode the two vectors, so the test is guaranteed to not be flaky.  (random in tests is usually a red flag)

Reply:
I changed `MakeVecs` to be deterministic and non-colliding.